src/Abstractions/NexusMods.Abstractions.NexusWebApi/IGameDomainToGameIdMappingCache.cs

```diff
@@ -0,0 +1,23 @@
+using DynamicData.Kernel;
+using NexusMods.Abstractions.NexusWebApi.Types.V2;
+using GameDomain = NexusMods.Abstractions.NexusWebApi.Types.GameDomain;
+namespace NexusMods.Abstractions.NexusWebApi;
+
+/// <summary>
+/// Caches the mapping between <see cref="GameDomain"/> and <see cref="GameId"/> values for fast lookup.
+/// Queries the API to populate missing values.
+/// </summary>
+public interface IGameDomainToGameIdMappingCache
+{
+    /// <summary>
+    /// Tries to get the corresponding <see cref="GameId"/>.
+    /// If there is no mapping, the API will be queried to get the mapping.
+    /// </summary>
+    ValueTask<Optional<GameId>> TryGetIdAsync(GameDomain gameDomain, CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Tries to get the corresponding <see cref="GameDomain"/>.
+    /// If there is no mapping, the API will be queried to get the mapping.
+    /// </summary>
+    ValueTask<Optional<GameDomain>> TryGetDomainAsync(GameId gameId, CancellationToken cancellationToken);
+}
```

Comment:
Since we'll still need the domains to create URLs in diagnostics and other places, it'll be better if this API is synchronous. Maybe on startup we could just fetch mappings for all games and then everything would be cached for the entirety that the app is running. The methods would then be super simple dictionary lookups.

That'd make this easier to use in diagnostics.

Reply:
We've been told that in the future the V2 API will require sign-in.
If the user is using the App for the first time, then doing it on startup is a no-go; unless you want to do it on login.

I did consider doing a synchronous API hack that just does `Task.Run().Result` on cache miss; since the results are persisted to MMDB across runs; I just figured it wasn't needed yet so I didn't want to be accused of overkilling it.

Reply:
Anyway, I added sync APIs that delegate to threadpool anyway, since you asked 😇